pkg/controller/volume/cache/desired_state_of_world.go

```diff
@@ -57,7 +57,7 @@ type DesiredStateOfWorld interface {
 	// should be attached to the specified node, the volume is implicitly added.
 	// If no node with the name nodeName exists in list of nodes managed by the
 	// attach/detach attached controller, an error is returned.
-	AddPod(podName types.UniquePodName, volumeSpec *volume.Spec, nodeName string) (api.UniqueVolumeName, error)
+	AddPod(pod *api.Pod, volumeSpec *volume.Spec, nodeName string) (api.UniqueVolumeName, error)
```

Comment:
Sure. But I am wondering whether we should use podName or podUID in the
parameter list to avoid confusion? If we want to use podUID, we have to
change other API parameter lists to make it consistent.

On Fri, Jun 17, 2016 at 5:31 PM, Saad Ali <notifications@github.com> wrote:

> In pkg/controller/volume/cache/desired_state_of_world.go
> <https://github.com/kubernetes/kubernetes/pull/27576#discussion_r67591159>
> :
>
> > @@ -57,7 +57,7 @@ type DesiredStateOfWorld interface {
> >  	// should be attached to the specified node, the volume is implicitly added.
> >  	// If no node with the name nodeName exists in list of nodes managed by the
> >  	// attach/detach attached controller, an error is returned.
> > -	AddPod(podName types.UniquePodName, volumeSpec *volume.Spec, nodeName string) (api.UniqueVolumeName, error)
> > +	AddPod(pod *api.Pod, volumeSpec *volume.Spec, nodeName string) (api.UniqueVolumeName, error)
>
> Sorry I was unclear. I'm ok with the pod object being passed in instead of
> individual pod fields, but please leave podName types.UniquePodName as
> is. I want the caller to be very explicit aware when calling this method
> what will be used as the key to refer to this pod.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/kubernetes/kubernetes/pull/27576/files/b106cab8833be12bff96236c139d950f5252137b#r67591159>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/ASSNxahzuIgUlCDF-5AqCALcOaZYvfiMks5qMzxNgaJpZM4I350f>
> .
>



-- 
- Jing